src/tests/intg/test_files_provider.py

```diff
@@ -0,0 +1,689 @@
+#
+# SSSD files domain tests
+#
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# This is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 only
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import stat
+import time
+import config
+import signal
+import subprocess
+import pytest
+import tempfile
+import shutil
+import pwd
+import grp
+
+import ent
+import sssd_id
+from sssd_nss import NssReturnCode
+from sssd_passwd import call_sssd_getpwnam, call_sssd_enumeration
+from sssd_group import call_sssd_getgrnam
+from files_ops import passwd_ops_setup, group_ops_setup
+from util import unindent
+
+# FIXME - remove before pushing
+from util import run_shell
+
+canary = dict(name='canary', passwd='x', uid=100001, gid=200001,
+              gecos='Used to check if passwd is resolvable',
+              dir='/home/canary',
+              shell='/bin/bash')
+
+user1 = dict(name='user1', passwd='x', uid=10001, gid=20001,
+             gecos='User for tests',
+             dir='/home/user1',
+             shell='/bin/bash')
+
+user2 = dict(name='user2', passwd='x', uid=10002, gid=20001,
+             gecos='User2 for tests',
+             dir='/home/user2',
+             shell='/bin/bash')
+
+canary_gr = dict(name='canary',
+                 gid=300001,
+                 mem=[])
+
+group1 = dict(name='group1',
+              gid=30001,
+              mem=['user1'])
+
+group12 = dict(name='group12',
+               gid=30012,
+               mem=['user1', 'user2'])
+
+group_nomem = dict(name='group_nomem',
+                   gid=40000,
+                   mem=[])
+
```

Comment:
And the same comment for gobal variables/ constants as above. lowercase -> uppercase